src/components/common/UserSwitchWrapper.js

```diff
@@ -0,0 +1,68 @@
+import React, { useContext } from 'react';
+import { useNavigate } from 'react-router';
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import { UserSwitchWrapper as GraaspUserSwitch } from '@graasp/ui';
+import { MUTATION_KEYS } from '@graasp/query-client';
+
+import { useMutation, hooks } from '../../config/queryClient';
+import { DOMAIN, SIGN_IN_PATH } from '../../config/constants';
+import { MEMBER_PROFILE_PATH } from '../../config/paths';
+import { CurrentUserContext } from '../context/CurrentUserContext';
+import {
+  buildMemberMenuItemId,
+  HEADER_MEMBER_MENU_BUTTON_ID,
+  HEADER_MEMBER_MENU_SEE_PROFILE_BUTTON_ID,
+  HEADER_MEMBER_MENU_SIGN_IN_BUTTON_ID,
+  HEADER_MEMBER_MENU_SIGN_OUT_BUTTON_ID,
+} from '../../config/selectors';
+
+const UserSwitchWrapper = ({ ButtonContent }) => {
+  const {
+    data: member,
+    isLoading,
+    isSuccess: isSuccessUser,
+  } = useContext(CurrentUserContext);
+  const navigate = useNavigate();
+  const { t } = useTranslation();
+  const { mutateAsync: signOut } = useMutation(MUTATION_KEYS.SIGN_OUT);
+  const { mutate: switchMember } = useMutation(MUTATION_KEYS.SWITCH_MEMBER);
+
+  return (
+    <>
+      <GraaspUserSwitch
+        ButtonContent={ButtonContent}
+        navigate={navigate}
+        signOut={signOut}
+        currentMember={member}
+        isCurrentMemberLoading={isLoading}
+        isCurrentMemberSuccess={isSuccessUser}
```

Comment:
I see that all values from `CurrentUserContext` are passed down as props. Is there a reason why `GraaspUserSwitch` is not invoking `useContext` itself? (I guess it is so that GraaspUserSwitch only has UI concerns). Though in which case, would it make sense for GraaspUserSwitch to take the context itself as props? The type could be restricted to the relevant props only from UI side. Not sure what's best 🤔

Reply:
`GraaspUserSwitch` is factored out in graasp-ui, so it's not aware of the context. At first the wrapper was in compose, but since I needed the exact same component I factored it out in graasp ui as well. Maybe I could have pass the context and use `useContext` in the component? Not sure it works.

Reply:
Ah my bad, my explanation was a bit confusing. Obviously I don't think passing the context object itself would be a great idea, but would it make sense to consume the context first and pass the value object as a whole to the `GraaspUserSwitch` component? Something like this:

```tsx
const userQuery = useContext(CurrentUserContext)

...
return (
   <>
      <GraaspUserSwitch
        ...
        currentMemberQuery={userQuery}
        ...
      />
   <>
)
```

Not sure if it's more readable though